contrib/src/main/java/com/datatorrent/contrib/parquet/ParquetFilePOJOReader.java

```diff
@@ -0,0 +1,318 @@
+  private static final Logger logger = LoggerFactory.getLogger(ParquetFilePOJOReader.class);
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> output = new DefaultOutputPort<Object>()
+  {
+
+    @Override
+    public void setup(Context.PortContext context)
+    {
+      pojoClass = context.getValue(Context.PortContext.TUPLE_CLASS);
+      pojoSetters = Maps.newHashMap();
+      for (Field f : pojoClass.getDeclaredFields()) {
+        try {
+          pojoSetters.put(f.getName(), generateSettersForField(pojoClass, f.getName()));
+        } catch (NoSuchFieldException | SecurityException e) {
+          logger.error("Failed to generate setters. Exception {}", e);
+          throw new RuntimeException(e);
+        }
+      }
+
+      if (groupToPOJOFieldsMapping == null) {
+        fieldInfos = createFieldInfoMap(generateFieldInfoInputs());
+      } else {
+        fieldInfos = createFieldInfoMap(groupToPOJOFieldsMapping);
+      }
+      initColumnFieldSetters();
+    }
+
+  };
+
+  /**
+   * Converts Group to POJO
+   */
+  @Override
+  protected Object convertGroup(Group group)
+  {
+    Object obj;
+    try {
+      obj = pojoClass.newInstance();
+    } catch (InstantiationException | IllegalAccessException ex) {
+      throw new RuntimeException(ex);
+    }
+    for (int i = 0; i < columnFieldSetters.size(); i++) {
+      try {
+        ParquetFilePOJOReader.ActiveFieldInfo afi = columnFieldSetters.get(i);
+        int fieldIndex = schema.getFieldIndex(afi.fieldInfo.getColumnName());
+        PrimitiveTypeName primitiveTypeName = schema.getType(fieldIndex).asPrimitiveType().getPrimitiveTypeName();
+
+        switch (primitiveTypeName) {
+
+          case BOOLEAN:
+            Boolean booleanVal = Boolean.parseBoolean(group.getValueToString(fieldIndex, 0));
+            pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, booleanVal);
+            break;
+
+          case INT32:
+            Integer intVal = Integer.parseInt(group.getValueToString(fieldIndex, 0));
+            pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, intVal);
+            break;
+
+          case INT64:
+            Long longVal = Long.parseLong(group.getValueToString(fieldIndex, 0));
+            pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, longVal);
+            break;
+
+          case FLOAT:
+            Float floatVal = Float.parseFloat(group.getValueToString(fieldIndex, 0));
+            pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, floatVal);
+            break;
+
+          case DOUBLE:
+            Double doubleVal = Double.parseDouble(group.getValueToString(fieldIndex, 0));
+            pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, doubleVal);
+            break;
+
+          case BINARY:
+            pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, group.getValueToString(fieldIndex, 0));
```

Comment:
Can we set this to byte[] instead... Setting toString may not be best thing to do here.